src/l1/mod.rs

```diff
@@ -256,13 +256,15 @@ impl InnerWatcher {
         if self.current_block > self.finalized_block {
             let finalized_block = self.get_finalized().await?;
 
-            self.finalized_block = finalized_block;
-            self.block_update_sender
-                .send(BlockUpdate::FinalityUpdate(finalized_block))
-                .await?;
-
-            self.unfinalized_blocks
-                .retain(|b| b.number > self.finalized_block)
+            // Only update finalized block if it has changed to avoid spamming the channel.
+            if self.finalized_block < finalized_block {
```

Comment:
nit: can you add a `tracing::info!` log here just so it's clear when this is triggered

Reply:
then it will flood the log with l1 blocks. i can add a debug tho